docs/content/en/docs/pipeline-stages/init.md

```diff
@@ -0,0 +1,171 @@
+
+If you have multiple `Dockerfile` or `build.gradle/pom.xml` files, Skaffold will provide an option
+to pair an image with one of the file.
+
+E.g.:                                                                                                                                                                                                                                                                                                                                 For a multi-services [microservices example](https://github.com/GoogleContainerTools/skaffold/tree/master/examples/microservices)
+```bash
+skaffold init
+? Choose the builder to build image gcr.io/k8s-skaffold/leeroy-app  [Use arrows to move, space to select, type to filter]
+> Docker (leeroy-app/Dockerfile)
+  Docker (leeroy-web/Dockerfile)
+  None (image not built from these sources)
+
+```
+
+
+{{< alert title="Note" >}}
+You can choose <code>None (image not built from these sources)</code> in case none of the suggested 
+options are correct. <br>
+You will have to manually set up build config for this artifact
+{{</alert>}}
+
+`skaffold` init also recognizes a maven or gradle project and will auto-suggest [`jib`]({{<relref "/docs/pipeline-stages/builders#jib-maven-and-gradle-locally">}}) builder.
+Currently `jib` artifact detection is disabled by default, you can turn it on using the flag `--XXenableJibInit`.
+
+You can try it this out on example [jib project](https://github.com/GoogleContainerTools/skaffold/tree/master/examples/jib-multimodule)
+```bash
+$cd examples/jib-multimodule
+$skaffold init --XXenableJibInit
+? Choose the builder to build image gcr.io/k8s-skaffold/skaffold-jib-1  [Use arrows to move, space to select, type to filter]
+> Jib Maven Plugin (skaffold-project-1, pom.xml)
+  Jib Maven Plugin (skaffold-project-2, pom.xml)
+  None (image not built from these sources)
+```
+
+
+In case you want to configure build artifacts on your own, use `--skip-build` flag.
+
+## Deploy Config Initialization
+`skaffold init` currently supports only [`Kubeclt` deployer]({{<relref "/docs/pipeline-stages/deployers#deploying-with-kubectl" >}})
+Skaffold will walk through all the `yaml` files in your project and find valid kubernetes manifest files.
+
+These files will be added to `deploy` config in `skaffold.yaml`.
+
+```yaml
+deploy:
+  kubectl:
+    manifests:
+    - leeroy-app/kubernetes/deployment.yaml
+    - leeroy-web/kubernetes/deployment.yaml
+```
+
+
+## Init API
+`skaffold init` also exposes an api which tools like IDEs can integrate with via flags.
+
+This API can be used to 
+
+1. Analyze a project workspace and discover all `Dockerfiles` and images.
+2. Generate Skaffold `build` config for a given artifact.
+
+**Init API contract**
+
+| API | flag | input/output |
+| ---- | --- | --- |
+| Analyze | `--analyze` and `--XXenableJibInit`| json encoded output of builders and images|  
+| Generate | `--artifact`| `=` delimited Dockerfile/image pair, or JSON string |
```

Comment:
can we show examples here? it's hard to understand the `=` delimited part or know what the JSON string format is.
```suggestion
| Generate | `--artifact`| "`=` delimited" build definition/image pair (for example: `=path1/Dockerfile=artifact1`) or JSON string (for example: ...) |
```